### PR TITLE
Update to use devenv

### LINKED ISSUE #218: Update Quarto development flow to use DevEnv
### Description
Introduce DevEnv to streamline the development workflow.

### Rationale
To enhance our documentation infrastructure, adopting [DevEnv](https://devenv.sh/) will standardize the development environment across all contributors. This ensures consistent dependencies, reduces on-boarding time, and simplifies the development process.

### Implementation
- Add devenv.nix and devenv.yaml configuration files to the repository.
- Update .gitignore to exclude any DevEnv-related generated files or directories.
- Revise the README to provide clear instructions on setting up and using DevEnv.

### PR DESCRIPTION
**Pull request by dweng0**
This PR adds DEVENV support to the repository. Included a section in the readme for installing and getting started

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,14 +2,24 @@
...
-- Clone this repository
-- In the root
-``` bash
-quarto render
-```
+### Setup
+
+- Install Nix using [nix-installer](https://zero-to-nix.com/start/install). The
+  official installer should also work, but may require additional configuration
+  that is not covered here.
```
Comment (jay-cousins) [inline README.md]:
```suggestion
- Install Nix using [nix-installer](https://zero-to-nix.com/start/install).
```

I'd just refer to installing  nix and avoid (a) installati may fail, (b) additional unspecified config may be required.

**Review by jay-cousins — CHANGES_REQUESTED** [review]
1 comment, else lgtm :)

File: README.md (hunk 1 above)
Comment (aiman) [reply README.md]: :+1:

Comment (dweng0) [reply README.md]: done

File: .gitignore [hunk 2]
```diff
@@ -6,3 +7,17 @@ docs
 /.luarc.json
 /_freeze/
 /_site/
+
+
+# Software Catalog
+__pycache__/
+
```
Comment (aiman) [inline .gitignore]: What is this for?

File: devenv.yaml
Comment (aiman) [inline devenv.yaml:1]: Bring back this file with the following content:

```
# yaml-language-server: $schema=https://devenv.sh/devenv.schema.json
inputs:
  nixpkgs:
    url: github:cachix/devenv-nixpkgs/rolling
```

**Review by aiman — CHANGES_REQUESTED** [review]
Please also add the `.envrc` file that devenv creates. It has the following content:

```
source_url "https://raw.githubusercontent.com/cachix/devenv/95f329d49a8a5289d31e0982652f7058a189bfca/direnvrc" "sha256-d+8cBpDfDBj41inrADaJt+bDWhOktwslgoP5YiGJ1v0="

use devenv
```

File: .gitignore (hunk 2 above)
Comment (dweng0) [reply .gitignore]: Removing. This is left over from when we were thinking about using python inside quarto to handle the rendering process.

File: devenv.yaml
Comment (dweng0) [reply devenv.yaml:1]: added this back in

**Comment by dweng0** [comment]
> Please also add the `.envrc` file that devenv creates. It has the following content:
> 
> ```
> source_url "https://raw.githubusercontent.com/cachix/devenv/95f329d49a8a5289d31e0982652f7058a189bfca/direnvrc" "sha256-d+8cBpDfDBj41inrADaJt+bDWhOktwslgoP5YiGJ1v0="
> 
> use devenv
> ```

updated

**Review by szemate — APPROVED** [review]

**Review by aiman — APPROVED** [review]
Great!

**Review by jay-cousins — APPROVED** [review]
LGTM :)

**Pull request merged**